rtloader/common/builtins/util.c

```diff
@@ -52,3 +61,42 @@ PyObject *headers(PyObject *self, PyObject *args, PyObject *kwargs)
 {
     return _public_headers(self, args, kwargs);
 }
+
+/*! \fn PyObject *obfuscate_sql(PyObject *self, PyObject *args)
+    \brief This function implements the `datadog_agent.obfuscate_sql` method, retrieving
+    the value for the key previously stored.
+    \param self A PyObject* pointer to the `datadog_agent` module.
+    \param args A PyObject* pointer to a tuple containing the key to retrieve.
+    \return A PyObject* pointer to the value.
+
+    This function is callable as the `datadog_agent.obfuscate_sql` Python method and
+    uses the `cb_obfuscate_sql()` callback to retrieve the value from the agent
+    with CGO. If the callback has not been set `None` will be returned.
+*/
+static PyObject *obfuscate_sql(PyObject *self, PyObject *args)
+{
+    // callback must be set
+    if (cb_obfuscate_sql == NULL) {
+        Py_RETURN_NONE;
+    }
+
+    char *key;
+
+    // datadog_agent.obfuscate_sql(key)
+    if (!PyArg_ParseTuple(args, "s", &key)) {
+        return NULL;
+    }
+
+    char *v = NULL;
+    // TODO: do we need Py_BEGIN_ALLOW_THREADS or PyGILState_Ensure?
+    v = cb_obfuscate_sql(key);
+
+    if (v == NULL) {
+        PyErr_SetString(PyExc_RuntimeError, "failed to read data");
```

Comment:
Should be `failed to obfuscate sql query`, no ?

Reply:
Yes updated to contain the obfuscation error.